lib/bolt/bolt_option_parser.rb

```diff
@@ -720,7 +720,7 @@ def initialize(options)
       define('--puppetfile FILEPATH',
              'Specify a Puppetfile to use when installing modules. (default: ~/.puppetlabs/bolt/Puppetfile)',
              'Modules are installed in the current Boltdir.') do |path|
-        @options[:puppetfile] = Pathname.new(File.expand_path(path))
+        @options[:puppetfile_path] = Pathname.new(File.expand_path(path))
```

Comment:
This is needed to get around the hash slicing used in `Bolt::Config`, since puppetfile config is under the `puppetfile` key.